src/client/src/StreamState.c

```diff
@@ -705,14 +705,34 @@ STATUS executeStoppedStreamState(UINT64 customData, UINT64 time)
     ENTERS();
     UNUSED_PARAM(time);
     STATUS retStatus = STATUS_SUCCESS;
+    UINT64 duration, viewByteSize;
     PKinesisVideoStream pKinesisVideoStream = STREAM_FROM_CUSTOM_DATA(customData);
 
     CHK(pKinesisVideoStream != NULL, STATUS_NULL_ARG);
 
     // Also store the connection stopping result
     pKinesisVideoStream->connectionDroppedResult = pKinesisVideoStream->base.result;
 
-    // Step the client state machine first
+    // Check if we want to prime the state machine based on whether we have any more content to send
+    // currently and if the error is a timeout.
+    if (SERVICE_CALL_RESULT_IS_A_TIMEOUT(pKinesisVideoStream->connectionDroppedResult)) {
+        CHK_STATUS(getAvailableViewSize(pKinesisVideoStream, &duration, &viewByteSize));
+        if (viewByteSize == 0) {
+            // Next time putFrame is called we will self-prime.
+            // The idea is to drop frames till new key frame which will
+            // become a stream start.
+            pKinesisVideoStream->resetGeneratorOnKeyFrame = TRUE;
+            pKinesisVideoStream->skipNonKeyFrames = TRUE;
+
+            // Set an indicator to step the state machine on new frame
+            pKinesisVideoStream->streamState = STREAM_STATE_NEW;
```

Comment:
if user has been streaming and then stopped, shouldnt it go to putStream state? when user start streaming again, it just open putMedia straight away.

Reply:
Excellent question. Right now, on a timeout it would go to STREAM_STATE_GET_ENDPOINT state. The existing code assumes that if the endpoint timed out then perhaps it's the best to get the new one. However, in case of intermittent producer, perhaps it makes sense to go directly to STREAM_STATE_PUT_STREAM state and if that fails with a timeout then go to the endpoint state. Let me make adjustments